pstoreds/addr_book.go

```diff
@@ -334,6 +335,8 @@ func (ab *dsAddrBook) GetPeerRecord(p peer.ID) *record.Envelope {
 		log.Errorf("unable to load record for peer %s: %v", p.Pretty(), err)
 		return nil
 	}
+	pr.RLock()
+	defer pr.RUnlock()
```

Comment:
we need a write lock if we are doing writes.

Reply:
@vyzo , looks to me that are only reads in this func.

Reply:
Just in case, the mentioned append is in another func which already does a write lock.

Reply:
ok, good.